### PR TITLE
[TASK] Prepare v12 compat

### PR DESCRIPTION
**Pull request by calien666**

### CODE REVIEW

File: Configuration/TCA/tx_securefilemount_folder.php [hunk 1]
```diff
@@ -9,10 +9,9 @@
         'iconfile' => 'EXT:secure_filemount/Resources/Public/Icons/Extension.svg',
         'tstamp' => 'tstamp',
         'crdate' => 'crdate',
-        'cruser_id' => 'cruser_id',
```
Comment (sbuerk) [inline Configuration/TCA/tx_securefilemount_folder.php:12]: Really remove it (for v11) ? In v12 this is stripped from core and handled over the history, but still it would make sense for v11. What's your decision on this on ?

File: composer.json [hunk 2]
```diff
@@ -16,9 +16,9 @@
 		}
 	],
 	"require": {
-		"php": "^7.4 || ^8.0",
-		"typo3/cms-core": "^11.5",
-		"typo3/cms-filelist": "^11.5"
+		"php": "^7.4 || ^8.0 || ^8.1 || ^8.2",
```
Comment (sbuerk) [inline composer.json:19]: php8.3 ?

File: ext_emconf.php [hunk 3]
```diff
@@ -11,8 +11,8 @@
     ],
     'constraints' => [
         'depends' => [
-            'typo3' => '11.5.0-11.5.99',
-            'php' => '7.4.0-8.1.99',
+            'typo3' => '11.5.0-12.4.99',
+            'php' => '7.4.0-8.2.99',
```
Comment (sbuerk) [inline ext_emconf.php:15]: php8.3 ?

File: composer.json [hunk 4]
```diff
@@ -16,9 +16,9 @@
...
+		"typo3/cms-core": "^11.5 || ^12.4",
+		"typo3/cms-filelist": "^11.5 || ^12.4"
```
Comment (sbuerk) [inline composer.json:21]: `typo3/cms-backend` and  `typo3/cms-frontend` should be added as required too

File: Classes/Domain/Repository/FolderRepository.php [hunk 5]
```diff
@@ -33,7 +34,7 @@ public function getFolder(\TYPO3\CMS\Core\Resource\Folder $folder): ?Folder
             ->where(
                 $db->expr()->eq(
                     'folder',
-                    $db->createNamedParameter($folder->getName())
+                    $db->createNamedParameter($folder->getIdentifier())
```
Comment (sbuerk) [inline Classes/Domain/Repository/FolderRepository.php:37]: This is not explained - neither in the commit message, PR message nor as comment in the code.

What's the reason for this change ? Is this safe for v11 AND v12 ? Is this a "bugfix" because being wrong beforehand - or is this related to a core change ?

Can you enlight me and others on that point ? Please :smile:

**Review by sbuerk — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (calien666) [reply composer.json:19]: could be

File: ext_emconf.php (hunk 3 above)
Comment (calien666) [reply ext_emconf.php:15]: your'e right

File: Classes/Domain/Repository/FolderRepository.php (hunk 5 above)
Comment (calien666) [reply Classes/Domain/Repository/FolderRepository.php:37]: getName only returned the name, but identifier returns full path. As name could be multiple, identifier is unique inside a storage. This is failsafe for getting the right storage folder.

File: Configuration/TCA/tx_securefilemount_folder.php (hunk 1 above)
Comment (calien666) [reply Configuration/TCA/tx_securefilemount_folder.php:12]: override in TCA/Overrides for Version compat in v11 done

**Review by sbuerk — APPROVED** [review]
Thanks for taking care of my comments. Nice job ! :heart: :+1:
